itm/dns_apps.go

```diff
@@ -76,28 +97,21 @@ func (s *dnsAppsServiceImpl) Create(opts *DNSAppOpts, publish bool) (*DNSApp, er
 			publishVal,
 		},
 	}
-	resp, err := s.client.post(dnsAppsBasePath, jsonOpts, qs)
+	resp, err := s.client.post(dnsAppsBasePath, opts.toJSON(), qs)
 	if err != nil {
-		log.Printf("Error issuing post request from DNSAppsServiceImpl.Create: %v", err)
+		log.Printf("[WARN] Error issuing post request from DNSAppsServiceImpl.Create: %v", err)
 		return nil, err
 	}
 	if 201 != resp.StatusCode {
-		return nil, &UnexpectedHTTPStatusError{
-			Expected: 201,
-			Got:      resp.StatusCode,
-		}
+		return nil, fmt.Errorf("Unexpected HTTP response. HTTP status code: %d. Body: %s", resp.StatusCode, resp.Body)
```

Comment:
Removed log level for consistency (log level goes on calls to `log.Printf` and `log.Println`).